etcdmgmt/etcd-mgmt.go

```diff
@@ -6,16 +6,65 @@ import (
 	"net/http"
 	"os"
 	"os/exec"
+	"strconv"
+	"syscall"
 	"time"
 
+	"github.com/gluster/glusterd2/utils"
+
 	log "github.com/Sirupsen/logrus"
 )
 
 // ExecName is to indicate the executable name, useful for mocking in tests
 var ExecName = "etcd"
 
-// StartEtcd () is to bring up etcd instance
-func StartEtcd() (*exec.Cmd, error) {
+func checkHealth(val time.Duration, listenClientUrls string) bool {
+	result := struct{ Health string }{}
+	// Checking health of etcd. Health of the etcd should be true,
+	// means etcd have initialized properly before using any etcd command
+	timer := time.NewTimer(time.Second * val)
+	for {
+		// Waiting for 15 second. Within 15 second health of etcd should
+		// be true otherwise it should throw an error
+		go func() {
+			<-timer.C
+			if result.Health != "true" {
+				log.Fatal("Health of etcd is not proper. Check etcd configuration.")
+			}
+		}()
+
+		resp, err := http.Get(listenClientUrls + "/health")
+		if err != nil {
+			continue
+		}
+
+		body, err := ioutil.ReadAll(resp.Body)
+
+		err = json.Unmarshal([]byte(body), &result)
+		if err != nil {
+			continue
+		}
+		if result.Health == "true" {
+			timer.Stop()
+			break
+		}
+	}
+	return true
+}
+
+var (
+	etcdPidDir  = "/var/run/gluster/"
```

Comment:
This cannot be hardcoded. If you are going to store the etcdPid in GD2's workdir, use config.WorkDir. Don't hardcode paths like this.